src/route-recognizer.js

```diff
@@ -135,13 +142,15 @@ export class RouteRecognizer {
         continue;
       }
 
-      output += '/';
       let segmentValue = segment.generate(routeParams, consumed);
-      if (segmentValue === null || segmentValue === undefined) {
-        throw new Error(`A value is required for route parameter '${segment.name}' in route '${name}'.`);
+      if (segmentValue == null) {
```

Comment:
I did not know there was a lint task, it is not mentioned in the readme 😕 
I reverted this specific change and generally made sure that there is no lint warning.
In my projects I usually configure the linter to accept double-equality with `null` (there is an option for that), which is only true for `undefined` and `null` and is a nice short syntax. I keep the rule that everything else is triple-equality, of course.

I also took the liberty to add the "no trailing whitespace" config inside `.editorconfig`, since the linter is also picky about that.